fluss-server/src/main/java/org/apache/fluss/server/metadata/TabletServerMetadataCache.java

```diff
@@ -155,9 +147,8 @@ public PartitionMetadata getPartitionMetadata(PhysicalTablePath partitionPath) {
                     partitionId,
                     new ArrayList<>(snapshot.getBucketMetadataForPartition(partitionId).values()));
         } else {
-            // TODO no need to get assignment from zk if refactor client metadata cache. Trace by
-            // https://github.com/apache/fluss/issues/483
-            return getPartitionMetadataFromZk(partitionPath, zkClient);
+
+            return null;
```

Comment:
Add `@Nullable` annotation to the return type of the method, or return `Optional` directly (it seems it is used as `Optional` later).